Logger/Student.cs

```diff
@@ -0,0 +1,49 @@
+﻿namespace Logger;
+
+public record Student : IEntity
+{
+    private FullName _name;
+    public int Sid { get; init; }
+
+    //TODO: explain why this is implemeted implicitly
+    public Guid Id { get; init; }
+
+    // implemented implicitly since Name is a frequently-accessed
+    public string Name
+    {
+        get
+        {
+            return _name.ToString();
+        }
+        set
+        {
+            _name = new(value);
+        }
+    }
+    public Student(FullName name, int sid, Guid id)
+    {
+        _name = name;
```

Comment:
```suggestion
        Name = name;
```
fields should be accessed through Properties. Comment applies to all places in file